packages/aws-cdk/lib/diff.ts

```diff
@@ -88,3 +93,7 @@ function buildLogicalToPathMap(template: cxapi.SynthesizedStack) {
   }
   return map;
 }
+
+function _hasAssets(stack: cxapi.SynthesizedStack) {
+  return Object.values(stack.metadata).find(entries => entries.find(entry => entry.type === cxapi.ASSET_METADATA) != null) != null;
```

Comment:
I am curious why you find it important to add `!= null`?

Reply:
The outer-most one is to force into a boolean context (for `tsc` to map it as a `boolean`), the middle one is out of habit.